Issue #120: QA 9-18

jahabeebs opened the pull request:
Closes #120 

## Description
- [X] Modal should close once connected 
- [X] Test token button overlapping. Change text to "Test Tokens 🪂" and reduce right padding to have even spacing
- [X] OD balance should be dollar currency formattted, commas and 2 decimals 
- [X] OD balance should be dollar currency formattted, commas and 2 decimals 
- [X] Footer is still not placed properly. The main page content height should be a minimum of screen height, so the footer never jumps up when there's no content

## Screenshots

Modal closes when connected

https://github.com/open-dollar/od-app/assets/47253537/ccdcfe5f-eacd-43c3-b6d3-6a632d346c6f

Test token button changes (note that the add token to wallet dropdown is now a little larger than the button itself to avoid the text wrapping to two lines--if you want I can just have it wrap)

https://github.com/open-dollar/od-app/assets/47253537/053d9c22-790e-4c21-a63f-9be9e17e5c96

Navbar has more responive layout when OD balance is large (note that OD is not formatted correctly in this example because I edited the text directly in the DOM to show it can be responsive with multiple #'s) 

https://github.com/open-dollar/od-app/assets/47253537/1415a649-4df8-402b-be0e-35e29000cd6d

OD balance is dollar currency formatted, commas & 2 decimals (example number is 112344)

<img width="1147" alt="input 112344" src="https://github.com/open-dollar/od-app/assets/47253537/32f914c9-9ef9-4a79-a9d2-e4334cb8cb8d">

OD balance is dollar currency formatted, commas & 2 decimals (example number is 10123)

<img width="1155" alt="input 10123" src="https://github.com/open-dollar/od-app/assets/47253537/2b81634a-20d8-470f-9c4d-a5b608d827c9">

Footer not jumping anymore (can compare with first video in this PR where footer is jumping on homepage)

https://github.com/open-dollar/od-app/assets/47253537/b00a9d57-99f3-4f37-af6c-fcc734a469db

pi0neerpat commented on src/components/Navbar.tsx:
```diff
@@ -2,9 +2,8 @@ import React, { useMemo, useState, useRef, useEffect } from 'react'
-import { utils } from 'ethers'
 
-import { formatNumber, getTokenLogo, newTransactionsFirst, returnWalletAddress } from '~/utils'
+import { formatDataNumber, getTokenLogo, newTransactionsFirst, returnWalletAddress } from '~/utils'
```
is formatNumber still being used in the app? Maybe worth removing if its no longer used to avoid confusion. I believe that was created by Iana, but maybe should have used `formatDataNumber` instead?

pi0neerpat commented on src/containers/Shared.tsx:
```diff
@@ -349,7 +349,9 @@ const Container = styled.div`
 
-const Content = styled.div``
+const Content = styled.div`
+    min-height: 100vh;
```
Good

pi0neerpat reviewed (approved):
Approved. Can you check that the dropdowns can be closed? I couldn't close them yesterday

jahabeebs commented:
> Approved. Can you check that the dropdowns can be closed? I couldn't close them yesterday

dropdowns are working

https://github.com/open-dollar/od-app/assets/47253537/847095f7-d749-407f-812d-910369b22819

I made these changes:
- renamed WALLET_CONNECT_PROJECT_ID to REACT_APP_WALLET_CONNECT_PROJECT_ID and realized I had the project ID hardcoded so I replaced that with the env variables and checked that it works in the deployment 
- globalDebtUtilization was crashing the analytics page--changed      
```               
globalDebtUtilization: transformToWadPercentage(result.globalDebt, result.globalDebtCeiling, 3)
```
to
```
transformToWadPercentage(result.globalDebt, result.globalDebtCeiling)
```
